### DISCUSSION TITLE
Lambda Container Image support

### DISCUSSION THREAD

**Discussion by 1davidmichael** [brefphp/bref#817] [General]
Hi,

I was wondering what the new [Container Image Support](https://aws.amazon.com/blogs/aws/new-for-aws-lambda-container-image-support/) for Lambda means for this project? I think it might make the published layers less needed but I believe some kind of runtime interface will still be needed.

I am looking at deploying at Laravel based app and am trying to keep costs low. If I could accomplish this via Lambda with container images it would be awesome.

**Comment by mnapoli** [comment]
Hi, check out [AWS Lambda can now run PHP using Docker Containers](https://mnapoli.fr/aws-lambda-php-docker-containers/) and especially this section:

---

### Which one to choose?

You may have understood now that "Container support" in Lambda mostly means that you can use Docker to package your application.

It does not involve radical architecture changes for us; our code runs the same way in Docker as in native runtimes: in an event-driven way.

It is a bit early to be definitive about this, but here is my recommendation as of now:

- use Lambda runtimes by default:
       - these are simpler to use and well documented
       - no Dockerfile to create (you don't even need to know Docker)
        with Bref, Lambda runtimes are automatically versioned in sync with the Bref version, making it much easier to keep up to date and upgrade
       - 3rd party tools support Lambda runtimes, most of them don't support containers yet
- use Docker if you have a specific reason for that:
       - you need to deploy an application larger than 250MB
       - you want control over the Linux image
       - you want to include an exotic PHP extension
       - you want to include specific system libraries or programs

---

Yes, you still need the PHP runtime client provided by Bref. Docker would mostly be a "deployment" mechanism, with a way to customize the system a bit more.

**Comment by akzincsystems** [comment]
@mnapoli Thanks for the feedback. I see you are ahead of the game as usual ;-)
